src/lib.rs

```diff
@@ -323,11 +346,14 @@ impl Signature {
 
     #[inline]
     /// Serializes the signature in compact format
-    pub fn serialize_compact<C>(&self, secp: &Secp256k1<C>) -> [u8; 64] {
+    pub fn serialize_compact(&self) -> [u8; 64] {
         let mut ret = [0; 64];
         unsafe {
-            let err = ffi::secp256k1_ecdsa_signature_serialize_compact(secp.ctx, ret.as_mut_ptr(),
-                                                                       self.as_ptr());
+            let err = ffi::secp256k1_ecdsa_signature_serialize_compact(
+                ffi::secp256k1_context_no_precomp,
+                ret.as_mut_ptr(),
+                self.as_ptr(),
+            );
             debug_assert!(err == 1);
```

Comment:
It's not in your patch so nothing to change here but I'll forget otherwise: any reason why we sometimes `debug_assert` and sometimes `assert`?

Reply:
I think `debug_assert` is used for sanity checks that won't be included in optimized builds...

Reply:
Because it's a sanity check. This function will never return a non-1 value.

Reply:
Actually I think every instance of `assert!` should be `debug_assert!` now.